src/vfs/real_fetcher.rs

```diff
@@ -176,13 +178,16 @@ impl VfsFetcher for RealFetcher {
         }
     }
 
-    fn watch(&mut self, path: &Path) {
+    fn watch(&self, path: &Path) {
         log::trace!("Watching path {}", path.display());
```

Comment:
This method locks twice. Does it matter from a performance perspective? I guess we could add a bulk watch if it matters?

Reply:
It's locking two different things that should be held on a short period, so it should be ok?